chalice/deploy/validate.py

```diff
@@ -7,6 +7,34 @@
 from chalice.config import Config  # noqa
 
 
+class ExperimentalFeatureError(Exception):
+    _ERROR_MSG = (
+        'You are using experimental features without explicitly opting in.\n'
+        'Experimental features do not guarantee backwards compatibility '
+        'and may be removed in the future.\n'
+        'If you still like to use these '
+        'experimental features, you can opt in by adding this to your '
+        'app.py file:\n\n%s\n\n'
+        'See https://chalice.readthedocs.io/en/latest/topics/experimental.rst'
+        ' for more details.'
+    )
+
+    def __init__(self, features_missing_opt_in):
+        # type: (Set[str]) -> None
+        self.features_missing_opt_in = features_missing_opt_in
+        msg = self._generate_msg(features_missing_opt_in)
+        super(ExperimentalFeatureError, self).__init__(msg)
+
+    def _generate_msg(self, missing_features):
+        # type: (Set[str]) -> str
+        opt_in_line = (
+            'app.experimental_feature_flags.update([\n'
+            '    %s\n'
```

Comment:
Nitpick but since this is all one line if we have a bunch of features this would generate a longer-than-80 line. Probably pretty easily. Also not hard to change to be one element per line so its always pep8.